### ISSUE TITLE
Bug report: Problem with poetry

### ISSUE THREAD

**Issue by MikeVL** [atomflunder/stringmatch#244]
Note: This error originates from the build backend, and is likely not a problem with poetry but with stringmatch (0.14.5) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"'.

`pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"`
```
Collecting stringmatch==0.14.5
  Downloading stringmatch-0.14.5.tar.gz (18 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      Traceback (most recent call last):
        File "/Users/mike/Library/Caches/pypoetry/virtualenvs/elec-WSF19ZTk-py3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/mike/Library/Caches/pypoetry/virtualenvs/elec-WSF19ZTk-py3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/mike/Library/Caches/pypoetry/virtualenvs/elec-WSF19ZTk-py3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/pip-build-env-78_sw9x3/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/pip-build-env-78_sw9x3/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/pip-build-env-78_sw9x3/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 13, in <module>
      FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
```

**Comment by atomflunder** [comment]
Hi @MikeVL , thanks for the bug report!

Unfortunately I cannot reproduce this error, neither with poetry nor with the given command. I tried both in Python 3.13, and Python 3.12.

```bash
poetry add stringmatch
Using version ^0.14.5 for stringmatch

Updating dependencies
Resolving dependencies... (1.0s)

Package operations: 6 installs, 0 updates, 0 removals

  - Installing mypy-extensions (1.0.0)
  - Installing typing-extensions (4.12.2)
  - Installing rapidfuzz (3.10.1)
  - Installing unidecode (1.3.8)
  - Installing mypy (1.13.0)
  - Installing stringmatch (0.14.5)

Writing lock file
```

and

```bash
pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"
Collecting stringmatch==0.14.5
  Downloading stringmatch-0.14.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (14 kB)
Collecting rapidfuzz==3.10.1 (from stringmatch==0.14.5)
  Downloading rapidfuzz-3.10.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)
Collecting unidecode==1.3.8 (from stringmatch==0.14.5)
  Downloading Unidecode-1.3.8-py3-none-any.whl.metadata (13 kB)
Collecting mypy==1.13.0 (from stringmatch==0.14.5)
  Downloading mypy-1.13.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_28_x86_64.whl.metadata (2.1 kB)
Collecting typing-extensions>=4.6.0 (from mypy==1.13.0->stringmatch==0.14.5)
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting mypy-extensions>=1.0.0 (from mypy==1.13.0->stringmatch==0.14.5)
  Downloading mypy_extensions-1.0.0-py3-none-any.whl.metadata (1.1 kB)
Downloading stringmatch-0.14.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (177 kB)
Downloading mypy-1.13.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_28_x86_64.whl (12.6 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.6/12.6 MB 6.8 MB/s eta 0:00:00
Downloading rapidfuzz-3.10.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 6.8 MB/s eta 0:00:00
Downloading Unidecode-1.3.8-py3-none-any.whl (235 kB)
Downloading mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Saved ./stringmatch-0.14.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
Saved ./mypy-1.13.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_28_x86_64.whl
Saved ./rapidfuzz-3.10.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
Saved ./Unidecode-1.3.8-py3-none-any.whl
Saved ./mypy_extensions-1.0.0-py3-none-any.whl
Saved ./typing_extensions-4.12.2-py3-none-any.whl
```

I tested this in a completely new, empty test repository, with only a "Hello World" main.py file, and this in the `pyproject.toml`:

```toml
[tool.poetry]
package-mode = false


[tool.poetry.dependencies]
python = "^3.9"
stringmatch = "^0.14.5"
```

Have you tried to verify this in a completely new environment or under different Python versions? I am quite puzzled why I can't reproduce the issue.  
Thanks in advance for your cooperation ☺️

**Comment by MikeVL** [comment]
Hi @atomflunder 

I'm trying to install in a new environment on a new computer with an APPLE M3 chip, on the old one with an Intel chip everything worked. Poetry installed from brew.


<img width="1715" alt="Снимок экрана 2024-12-28 в 14 03 20" src="https://github.com/user-attachments/assets/a6e4292f-5f07-40b5-8930-a852e8078873" />

**Comment by MikeVL** [comment]
```
uv pip install stringmatch
Resolved 6 packages in 1.10s
  × Failed to build `stringmatch==0.14.5`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stderr]
      Traceback (most recent call last):
        File "<string>", line 14, in <module>
        File "/Users/mike/.cache/uv/builds-v0/.tmppgEkBw/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/mike/.cache/uv/builds-v0/.tmppgEkBw/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/Users/mike/.cache/uv/builds-v0/.tmppgEkBw/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 13, in <module>
      FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'

      hint: This usually indicates a problem with the package or the build environment.
```

**Comment by atomflunder** [comment]
@MikeVL 

I don't have a ARM-based Mac unfortunately do check the issue myself, but I think I may have a clue on what's going on.

I used a very outdated workflow with MacOS 12, and it did not provide any pre-built wheels for MacOS ARM devices at all, only for Mac x86 based devices, that's why it used to work with the Intel chips.

I updated the workflow so that it works with MacOS 14, which seems like it's only ARM based, so it only generates ARM wheels. I may have to figure out how to support both.

Anyways, I uploaded the newly generated ARM wheels to PyPI, still stringmatch v0.14.5: https://pypi.org/manage/project/stringmatch/release/0.14.5/

Can you try to install it now?

**Comment by MikeVL** [comment]
```
pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"

Collecting stringmatch==0.14.5
  Downloading stringmatch-0.14.5-cp312-cp312-macosx_11_0_arm64.whl.metadata (14 kB)
Collecting rapidfuzz==3.11.0 (from stringmatch==0.14.5)
  Downloading rapidfuzz-3.11.0-cp312-cp312-macosx_11_0_arm64.whl.metadata (11 kB)
Collecting unidecode==1.3.8 (from stringmatch==0.14.5)
  Downloading Unidecode-1.3.8-py3-none-any.whl.metadata (13 kB)
Collecting mypy==1.14.0 (from stringmatch==0.14.5)
  Downloading mypy-1.14.0-cp312-cp312-macosx_11_0_arm64.whl.metadata (2.1 kB)
Collecting typing_extensions>=4.6.0 (from mypy==1.14.0->stringmatch==0.14.5)
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting mypy_extensions>=1.0.0 (from mypy==1.14.0->stringmatch==0.14.5)
  Downloading mypy_extensions-1.0.0-py3-none-any.whl.metadata (1.1 kB)
Downloading stringmatch-0.14.5-cp312-cp312-macosx_11_0_arm64.whl (93 kB)
Downloading mypy-1.14.0-cp312-cp312-macosx_11_0_arm64.whl (10.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.3/10.3 MB 8.4 MB/s eta 0:00:00
Downloading rapidfuzz-3.11.0-cp312-cp312-macosx_11_0_arm64.whl (1.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 10.8 MB/s eta 0:00:00
Downloading Unidecode-1.3.8-py3-none-any.whl (235 kB)
Downloading mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Saved ./stringmatch-0.14.5-cp312-cp312-macosx_11_0_arm64.whl
Saved ./mypy-1.14.0-cp312-cp312-macosx_11_0_arm64.whl
Saved ./rapidfuzz-3.11.0-cp312-cp312-macosx_11_0_arm64.whl
Saved ./Unidecode-1.3.8-py3-none-any.whl
Saved ./mypy_extensions-1.0.0-py3-none-any.whl
Saved ./typing_extensions-4.12.2-py3-none-any.whl
```

**Comment by atomflunder** [comment]
@MikeVL Cool, looks good 👍 Thank you for your help in troubleshooting!

If you are okay with it, I would close this issue and then figure out the rest of the details mentioned in my previous comment.

**Comment by MikeVL** [comment]
Installing via poetry:

> Installing stringmatch (0.14.5): Failed

**Comment by MikeVL** [comment]
```
  - Installing stringmatch (0.14.5): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke get_requires_for_build_wheel

  Traceback (most recent call last):
    File "/opt/homebrew/Cellar/poetry/1.8.5/libexec/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
      main()
    File "/opt/homebrew/Cellar/poetry/1.8.5/libexec/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in main
      json_out["return_val"] = hook(**hook_input["kwargs"])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/homebrew/Cellar/poetry/1.8.5/libexec/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 143, in get_requires_for_build_wheel
      return hook(config_settings)
             ^^^^^^^^^^^^^^^^^^^^^
    File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/tmpv5xx0p4z/.venv/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=[])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/tmpv5xx0p4z/.venv/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
      self.run_setup()
    File "/private/var/folders/0n/1kbw3lrx25x13vnhdtyf65wh0000gn/T/tmpv5xx0p4z/.venv/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
      exec(code, locals())
    File "<string>", line 13, in <module>
  FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'


  at /opt/homebrew/Cellar/poetry/1.8.5/libexec/lib/python3.13/site-packages/poetry/installation/chef.py:164 in _prepare
      160│
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      162│
      163│             if error is not None:
    → 164│                 raise error from None
      165│
      166│             return path
      167│
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with stringmatch (0.14.5) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"'.
```

```
✗ pip wheel --no-cache-dir --use-pep517 "stringmatch (==0.14.5)"
Collecting stringmatch==0.14.5
  File was already downloaded /Users/mike/elec/stringmatch-0.14.5-cp312-cp312-macosx_11_0_arm64.whl
Collecting rapidfuzz==3.11.0 (from stringmatch==0.14.5)
  File was already downloaded /Users/mike/elec/rapidfuzz-3.11.0-cp312-cp312-macosx_11_0_arm64.whl
Collecting unidecode==1.3.8 (from stringmatch==0.14.5)
  File was already downloaded /Users/mike/elec/Unidecode-1.3.8-py3-none-any.whl
Collecting mypy==1.14.0 (from stringmatch==0.14.5)
  File was already downloaded /Users/mike/elec/mypy-1.14.0-cp312-cp312-macosx_11_0_arm64.whl
Collecting typing_extensions>=4.6.0 (from mypy==1.14.0->stringmatch==0.14.5)
  File was already downloaded /Users/mike/elec/typing_extensions-4.12.2-py3-none-any.whl
Collecting mypy_extensions>=1.0.0 (from mypy==1.14.0->stringmatch==0.14.5)
  File was already downloaded /Users/mike/elec/mypy_extensions-1.0.0-py3-none-any.whl
```

**Comment by atomflunder** [comment]
Ah then nevermind. Is it maybe some kind of caching issue? Could you try to clear the poetry cache and then try again?  
Or maybe I need to push out a new release?

**Comment by MikeVL** [comment]
Yes, I cleared the cache, but maybe it's still there somewhere

**Comment by MikeVL** [comment]
Also poetry 
```
poetry install --no-cache
Installing dependencies from lock file

Package operations: 1 install, 2 updates, 0 removals

  - Downgrading mypy (1.14.0 -> 1.13.0)
  - Downgrading rapidfuzz (3.11.0 -> 3.10.1)
  - Installing stringmatch (0.14.5)
  ```

**Comment by atomflunder** [comment]
Hmm okay, I'll try to push out a new minor version then and see if that solves it.  
Building all of the wheels for a new version usually does take the Github Action around 2 hours though, as a heads up.

**Comment by atomflunder** [comment]
@MikeVL Okay, I published v0.14.6 on PyPI. Let me know if it works or not 🙂  

```bash
poetry add stringmatch@0.14.6 --no-cache

Updating dependencies
Resolving dependencies... (1.1s)

Package operations: 0 installs, 3 updates, 0 removals

  - Updating mypy (1.13.0 -> 1.14.0)
  - Updating rapidfuzz (3.10.1 -> 3.11.0)
  - Updating stringmatch (0.14.5 -> 0.14.6)

Writing lock file
```
